_posts/2020-04-27-feasibility-discv5.md

```diff
@@ -0,0 +1,70 @@
+
+## Motivating Problem
+
+The discovery method currently used by [Status](https://status.im), is made up of various components and grew over time to solve a mix of problems. We want to find a simplify this while maintaining some of the properties we currently have.
+
+Namely, we want to ensure censorship resistance to state-level adversaries. One of the issues Status had which caused us them add to their discovery method was the fact that addresses from providers like AWS and GCP were blocked both in Russia and China. Additionally, one of the main factors required is the ability to function on resource restricted devices.
+
+Considering we are talking about resource restricted devices, let's look at the implications and what we need to consider:
+
+ - **Battery consumption** - constant connections like websockets consume a lot of battery life.
+ - **CPU usage** - certain discovery methods may be CPU incentive, slowing an app down and making it unusable.
+ - **Bandwidth consumption** - a lot of users will be using data plans, the discovery method needs to be efficient in order to accommodate those users without using up significant portions of their data plans.
+ - **Short connection windows** - the discovery algorithm needs to be low latency, that means it needs to return results fast. This is because many users will only have the app open for a short amount of time.
+ - **Not publicly connectable** - There is a good chance that most resource restricted devices are not publicly connectable.
+
+For a node to be able to participate as both a provider, and a consumer in the discovery method. Meaning a node both reads from other nodes' stored DHTs and hosts the DHT for other nodes to read from, it needs to be publically connectable. This means another node must be able to connect to some public IP of the given node. 
+
+With devices that are behind a NAT, this is easier said than done. Especially mobile devices, that when connected to 4G LTE networks are often stuck behind a symmetric NAT, drastically reducing the the succeess rate of NAT traversal. Keeping this in mind, it becomes obvious that most resource restricted devices will be consumers rather than providers due to this technical limitation.
+
+## Overview
+
+The main things we wanted to investigate was the overhead on finding a peer. This means we wanted to look at both the bandwidth, latency and effectiveness of this. There are 2 methods which we can use to find a peer:
+ - We can find a peer with a specific ID, using normal lookup methods as documented by Kademlia.
+ - We can find a peer that advertises a capability, this is possible using either capabilities advertised in the ENR or through [topic tables](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-theory.md#topic-advertisement).
+
+## Feasbility
```

Comment:
Maybe add a section here on finding node being OK and Node ENR being OK assuming concentration is reasonable